### DISCUSSION TITLE
Eliminating the Close Window "X" for dBase on dosemu2

### DISCUSSION THREAD

**Discussion by WayCon2** [dosemu2/dosemu2#1823] [Q&A]
Hello, 
Everything is rosy when I'm running dBase IV on dosemu2, but I would like to prevent anyone from closing dBase via the terminal window "X" in the upper right hand corner. Closing that way results in having to re-index everything, which I would like to avoid. I note that starting dosemu2 from Terminal eliminates that "X" with: 
$ dosemu -t 
The resulting warning is good enough for my purposes. But that command doesn't seem to work from my little bash file. What is working for me that opens dBase is the following line: 
flock -w .007 /home/wayne/.dosemu/drive_c/testtext.txt -c dosemu; 
The "flock" command is there to prevent two versions of dosemu2 from running at the same time. Does anyone have any suggestion? 
In any case, I thank you for your time. 
Sincerely, 
WayCon2

**Comment by andrewbird** [comment]
I'm not exactly sure why your script is not working, but it could be that the `-t` is not being passed to dosemu but instead is being given to flock command. You might need to try quoting something like `-c "dosemu -t"`.

That being said if you are using X there is another way, try adding `$_X_noclose = (on)` to your .dosemurc, which should remove the close button. Unfortunately I didn't figure out a way to do it in SDL mode, so you'll probably have to launch dosemu as `dosemu -X`.

Hope it helps.

**Comment by WayCon2** [comment]
Thank you, andrewbird, 
I appreciate that you got back to me so quickly. What worked for me, based on your suggestions, was: 
flock -w .007 /home/wayne/.dosemu/drive_c/testtext.txt -c "dosemu -X"; 
Just right. I appreciate your suggestions.
SIncerely,
WayCon2

**Comment by WayCon2** [comment]
Hello, again, andrewbird, 
Here's an update: the flock line that worked, above, only eliminates the "X" when the file /etc/dosemu/dosemu.conf contains the change that you suggested, which is: 
$_X_noclose = (on) 
The two changes, together, work. Strange world, but now it's documented. 
Again, thank you, 
Sincerely, 
WayCon2

**Reply by andrewbird** [reply]
Yes, the `-X` switches the graphics window to use the X windows functions directly rather than doing it via SDL, the conf change eliminates the close button but only in X Windows mode, hence the two are both required.
